compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -3916,19 +3933,17 @@ class JSCodeGen()(using genCtx: Context) {
       }
 
       (f, true)
-    } else /*if (jsInterop.topLevelExportsOf(sym).nonEmpty) {
-      val f = js.SelectStatic(encodeClassName(sym.owner),
-          encodeFieldSym(sym))(jstpe.AnyType)
+    } else if (sym.hasAnnotation(jsdefn.JSExportTopLevelAnnot)) {
```

Comment:
Do you recall why we didn't simply do this in Scala 2?

Reply:
I don't know. It's probably historical, following a long chain of changes. `@JSExportTopLevel` inherited from `@JSExport` on classes and objects, in which the name could be implicit; which might explain it. git blame suggests https://github.com/scala-js/scala-js/commit/273e9b82fea10628e4669aba121027b07c34f5e0 as the moment at which we introduced those out-of-the-band mutable maps, but at the time `@JSExportTopLevel` did not exist yet.

We could try to get rid of it in Scala 2.